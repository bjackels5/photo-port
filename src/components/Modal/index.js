/*
Modal Component
*/

import React from 'react';

const Modal = ({onClose, currentPhoto}) => {
    const { name, category, description, index} = currentPhoto;

    return (
        <div className = "modalBackground">
            <div className = "modalContainer">
                <h3 className="ModalTitle">{name}</h3>
                <img alt="current category" src={require(`../../assets/large/${category}/${index}.jpg`).default} />
                <p> {description} </p>
                <button type="button" onClick={onClose}>Close</button>
            </div>
        </div>        
    )
}

export default Modal;