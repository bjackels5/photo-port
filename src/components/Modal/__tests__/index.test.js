import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '../'

const mockToggleModal = jest.fn();


afterEach(cleanup);

const currentPhoto = {
    name: "Park bench",
    category: 'landscape',
    description: 'jflsjljaopi jfoaji asdjfoiuhjwjwerjmcvosi',
    index: 1
}

describe('Modal is rendering', () => {
  it('renders', () => {
    render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto} />);
  });

  it('renders', () => {
    const { asFragment } = render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto} />)
    expect(asFragment()).toMatchSnapshot()
  });
});

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal onClose={mockToggleModal} currentPhoto={currentPhoto} />);
        fireEvent.click(getByText('Close'));
        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });    
})