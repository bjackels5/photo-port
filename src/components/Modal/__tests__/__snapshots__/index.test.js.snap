// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering renders 1`] = `
<DocumentFragment>
  <div
    class="modalBackground"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="ModalTitle"
      >
        Park bench
      </h3>
      <img
        alt="current category"
      />
      <p>
         jflsjljaopi jfoaji asdjfoiuhjwjwerjmcvosi 
      </p>
      <button
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
